vmngclient/tests/test_mtt_aaa_api.py

```diff
@@ -0,0 +1,297 @@
+import unittest
+from unittest.mock import patch
+
+from parameterized import parameterized  # type: ignore
+
+from vmngclient.api.mtt_aaa_api import TenantAaaAPI, TenantRadiusAPI, TenantTacacsAPI
+from vmngclient.dataclasses import TenantAAA, TenantRadiusServer, TenantTacacsServer
+from vmngclient.utils.creation_tools import create_dataclass
+
+
+class TestAaaAPI(unittest.TestCase):
+    def setUp(self) -> None:
+        self.aaa = {
+            "authOrder": ["radius", "tacacs", "local"],
+            "authFallback": True,
+            "adminAuthOrder": True,
+            "auditDisable": False,
+            "accounting": False,
+            "radiusServers": "server1",
+        }
+        self.p_aaa = {
+            "authOrder": ["radius", "local"],
+            "authFallback": True,
+            "adminAuthOrder": True,
+            "auditDisable": True,
+            "accounting": True,
+            "radiusServers": "serverX",
+        }
+        self.aaa_dataclass = create_dataclass(TenantAAA, self.aaa)
+        self.p_aaa_dataclass = create_dataclass(TenantAAA, self.p_aaa)
+
+    @parameterized.expand([[200, True], [400, False]])
+    @patch("vmngclient.session.Session")
+    @patch("requests.Response")
+    def test_del_aaa(self, status_code, expected_outcome, mock_session, mock_response):
+        # Arrange
+        mock_session.get_data.return_value = self.aaa
+        mock_session.delete.return_value = mock_response
+        mock_response.status_code = status_code
+        # Act
+        answer = TenantAaaAPI(mock_session).del_aaa()
+        # Assert
+        self.assertEqual(answer, expected_outcome)
+
+    @parameterized.expand([[200, True], [400, False]])
+    @patch("vmngclient.session.Session")
+    @patch("requests.Response")
+    def test_add_aaa(self, status_code, expected_outcome, mock_session, mock_response):
+        # Arrange
+        mock_session.get_data.return_value = self.aaa
+        mock_session.post.return_value = mock_response
+        mock_response.status_code = status_code
+        # Act
+        answer = TenantAaaAPI(mock_session).add_aaa(self.aaa_dataclass)
```

Comment:
```suggestion
        answer = TenantAaaAPI(mock_session).add(self.aaa_dataclass)
```
simplified method name looks better

Reply:
there are three components here: AAA, RADIUS and TACACS. Thats why added respective components.